src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java

```diff
@@ -369,6 +386,21 @@ private AuthConfig getAuthConfigFromECSTaskRole() throws IOException {
         }
     }
 
+    /** This is present to support testing */
+    private Map<String, Object> getMetadataSettings(Settings settings) {
+        Server server = settings.getServer("junit.ecs-meta");
```

Comment:
Could we avoid to refer to tests from the code-to-test ? 

I really appreciate the effort put into testing (although its more an integration then a unit-test, but I'm not dogmatic here except if the test duration increase significantly).

I suggest that instead of going over the settings, just use an environment variable like `ECS_AUTH_SERVICE=<host>:<port>` which is just set prior to the test, and, if not set, uses the ECS default values. That way you avoid a coupling to the test (and provide kind of an easy to use general feature, which we might expose to others, but wouldn't comment it yet.)

Reply:
Sure can do that.
In that case I'd suggest to integrate [system-rules](https://github.com/stefanbirkner/system-rules) in order to manipulate the [Environment Variables](https://stefanbirkner.github.io/system-rules/#EnvironmentVariables)

Reply:
I agree that this is more of an integration test; but this code is really hard to test without extracting methods for the sake of testing that than have to be mocked/overridden (the part where the service is called).
I can do that as well; not my personal preference but it's your project :wink:

Reply:
> In that case I'd suggest to integrate [system-rules](https://github.com/stefanbirkner/system-rules) in order to manipulate the [Environment Variables](https://stefanbirkner.github.io/system-rules/#EnvironmentVariables)

Sounds good.

> I agree that this is more of an integration test; but this code is really hard to test without extracting methods for the sake of testing that than have to be mocked/overridden (the part where the service is called).

No, I'm fine with using an HTTP server here, no worries.

Reply:
Great, I'll update the PR then...